packages/blockchain-extension/extension/commands/importNodesToEnvironmentCommand.ts

```diff
@@ -128,32 +129,44 @@ export async function importNodesToEnvironment(environmentRegistryEntry: FabricE
                 }
             }
         } else {
-            const keytar: any = ModuleUtil.getCoreNodeModule('keytar');
-            if (!keytar) {
-                throw new Error('Error importing the keytar module');
-            }
-
             const GET_ALL_COMPONENTS: string = '/ak/api/v1/components';
             const api: string = environmentRegistryEntry.url.replace(/\/$/, '') + GET_ALL_COMPONENTS;
             let response: any;
+            let requestOptions: any;
 
+            // establish connection to Ops Tools
             try {
-                // establish connection to Ops Tools and retrieve json file
-                const credentials: string = await keytar.getPassword('blockchain-vscode-ext', environmentRegistryEntry.url);
-                const credentialsArray: string[] = credentials.split(':');
-                if (credentialsArray.length !== 3) {
-                    throw new Error(`Unable to retrieve the stored credentials`);
+                if (environmentRegistryEntry.environmentType === EnvironmentType.SAAS_OPS_TOOLS_ENVIRONMENT) {
+                    const accessToken: string = await ExtensionsInteractionUtil.cloudAccountGetAccessToken();
+                    if (!accessToken) {
+                        return;
+                    }
+                    requestOptions = { headers: { Authorization: `Bearer ${accessToken}` } };
+                } else {
+                    const keytar: any = ModuleUtil.getCoreNodeModule('keytar');
+                    if (!keytar) {
+                        throw new Error('Error importing the keytar module');
+                    }
+
+                    const credentials: string = await keytar.getPassword('blockchain-vscode-ext', environmentRegistryEntry.url);
+                    const credentialsArray: string[] = credentials.split(':');
```

Comment:
Just wondering where you've done 'splits', if its worth writing a comment that says what you're expecting the output to be? Might make it easier to debug if Cloud changes the values/format.

E.g. `const credentialsArray: string[] = credentials.split(':'); // Format ['hello', 'world', 'somethingElse']`

Don't know - what do you think?

Reply:
The format is decided by us in 'addEnvironmentCommand', so our extension is in control - there will be no changes unless we make them. Happy to add a comment for readability though.